ReferenceAppKotlin/app/src/main/java/com/android/tv/reference/watchnext/VideoToEngageEntityConverter.kt

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.tv.reference.watchnext
+
+import android.media.tv.TvContentRating
+import android.net.Uri
+import androidx.tvprovider.media.tv.TvContractCompat.WatchNextPrograms
+import com.android.tv.reference.shared.datamodel.Video
+import com.android.tv.reference.shared.datamodel.VideoType
+import com.android.tv.reference.watchnext.EngageWatchNextService.Companion.WatchNextVideo
+import com.google.android.engage.common.datamodel.ContentAvailability
+import com.google.android.engage.common.datamodel.Image
+import com.google.android.engage.video.datamodel.MovieEntity
+import com.google.android.engage.video.datamodel.RatingSystem
+import com.google.android.engage.video.datamodel.TvEpisodeEntity
+import com.google.android.engage.video.datamodel.VideoEntity
+import com.google.android.engage.video.datamodel.WatchNextType
+
+object VideoToEngageEntityConverter {
```

Comment:
This object is useful for developers as it shows how they can convert the data models for Client Side Play Next to the ones for Video Discovery At the same time, I am wondering if we should remove all Client Side Play Next code and replace them with the ones for Video Discovery API.

Reply:
Let's keep the current implementation as is.